tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,361 @@
+import pytest
+
+import time
+import logging
+import tempfile
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload
+from scapy.all import Packet, ByteField, ShortField, MACField, XStrFixedLenField, ConditionalField
+from scapy.all import split_layers, bind_layers, rdpcap
+import scapy.contrib.lacp
+import scapy.layers.l2
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology("t0", "t1")
+]
+
+
+class LACPRetryCount(Packet):
+    name = "LACPRetryCount"
+    fields_desc = [
+        ByteField("version", 0xf1),
+        ByteField("actor_type", 1),
+        ByteField("actor_length", 20),
+        ShortField("actor_system_priority", 0),
+        MACField("actor_system", None),
+        ShortField("actor_key", 0),
+        ShortField("actor_port_priority", 0),
+        ShortField("actor_port_number", 0),
+        ByteField("actor_state", 0),
+        XStrFixedLenField("actor_reserved", "", 3),
+        ByteField("partner_type", 2),
+        ByteField("partner_length", 20),
+        ShortField("partner_system_priority", 0),
+        MACField("partner_system", None),
+        ShortField("partner_key", 0),
+        ShortField("partner_port_priority", 0),
+        ShortField("partner_port_number", 0),
+        ByteField("partner_state", 0),
+        XStrFixedLenField("partner_reserved", "", 3),
+        ByteField("collector_type", 3),
+        ByteField("collector_length", 16),
+        ShortField("collector_max_delay", 0),
+        XStrFixedLenField("collector_reserved", "", 12),
+        ConditionalField(ByteField("actor_retry_count_type", 0x80), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("actor_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("actor_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("actor_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_type", 0x81), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("partner_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ByteField("terminator_type", 0),
+        ByteField("terminator_length", 0),
+        ConditionalField(XStrFixedLenField("reserved", "", 42), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("reserved", "", 50), lambda pkt:pkt.version != 0xf1),
+    ]
+
+
+@pytest.fixture(scope="class")
```

Comment:
Since the scope is `class` would the setting retry-count to 5 happen only once? I think the expectation is for that to happen per case within that class.

Reply:
Yes, the flow would be:
1. Set the retry count to 5
2. Run applicable tests that use this fixture
3. Reset the retry count to 3

This is largely done for time purposes; resetting the retry count to 3 takes 60 seconds for it to actually take effect, and having this be run per-testcase adds 60 seconds to the runtime for each test case.